### ISSUE TITLE
Page gets redirected, if the page URL is the same as portal alias with a different domain

### ISSUE THREAD

**Issue by berkarslan-xo** [dnnsoftware/Dnn.Platform#4669]
## Description of bug
If the page name is the same as the portal name in a different domain, the page can't be accessed since it redirects to the other portal when tried to be accessed.

## Steps to reproduce
1. Set a DNN site with the following domain: "**domain**.dnndev.me"
2. Create a page without a parent page and name it "test1" (URL to access the page should be "**domain**.dnndev.me/test1")
3. Create a new portal using the same page name as the site name ("test1"). The URL should be "dnndev.me/test1".
4. Try to access the test1 page again.

## Current behavior
The request gets redirected to the new portal's home page.

## Expected behavior
The "test1" page of the "**domain**.dnndev.me" site should be opened. 

## Screenshots
Reproduction video: https://drive.google.com/file/d/1syAoLyUFDKJumYP5ImqX7bkTJeaXP1y1/view

## Affected version

* [X] 10.00.00 alpha build
* [ ] 09.09.00 latest supported release

## Affected browser

* [x] Chrome
* [ ] Firefox
* [ ] Safari
* [ ] Internet Explorer 11
* [ ] Microsoft Edge (Classic)
* [ ] Microsoft Edge Chromium

**Comment by sleupold** [comment]
the problem seems to be in the routine to create a sub-website, which needs to check for an existing page path and should prevent the site to be created in this case (AFAIK creating a page URL doesn't work, if there already is a sub-website with same address.)

**Comment by sleupold** [comment]
btw. I disagree that a page URL should override a sub-website with same URL.
Is there any specific use case you are having in mind?

**Comment by berkarslan-xo** [comment]
Github applied a formatting to an important text there and I fixed it and it might have created this confusion. I edited the following:

![image](https://user-images.githubusercontent.com/45750757/119052295-0e0b7c00-b9cd-11eb-805c-7c938aef4a3e.png)

To explain the issue in more detail: If there is a site with "test" name for example, you can't have a page named as "test" (in the same site) and there is already a check for that. However, the problem occurs only when the portal gets created on a different domain. Because the following URLs should return different pages:

- domain.dnndev.me/test1 (refers to page)
- dnndev.me/test1 (refers to a site's home page)

You can check my PR from https://github.com/dnnsoftware/Dnn.Platform/pull/4670 btw.

**Comment by sleupold** [comment]
there is a problem caused by the redirect for sub-portals.
If I remember correctly, if you create a sub-website "domain.dnndev.me/test1", a folder is created in the root folder (/test1) with a redirecting default.aspx. This should be accessible with higher priority than a page with same name in the parent site.
However, UI of DNN should try to avoid sub-website and page with same path to be created.

**Comment by berkarslan-xo** [comment]
Asking for additional confirmations that this can't be implemented before closing the issue.

**Comment by berkarslan-xo** [comment]
Hi @bdukes . The issue is not related to subdirectory creation in any way.

The your last paragraph you stated that for the ambiguity cases, the alias should take the precedence. However, that case is not possible at all because if you have a site with example.com/dallas URL, you can't have a page named "dallas" in the same domain because there is already a validation for that:
![image](https://user-images.githubusercontent.com/45750757/119160318-cf270600-ba60-11eb-8ddc-aa4926a5f2e3.png)
and vice versa (creating the page first and then the site):
![image](https://user-images.githubusercontent.com/45750757/119161522-16fa5d00-ba62-11eb-9345-775fbca9df58.png)

The problem with the code was that it wasn't checking the pages at all and it was returning directly in case there is a possible match in the alias and my code fixes that. I can provide detailed debugging pictures etc. if you require, just let me know.

**Comment by sleupold** [comment]
As far as I can see, PR #4670 does contain some discussion. 
Did you check out feedback from @bdukes ?

**Comment by v-karbovnichy** [comment]
yes, the conversation there ended by redirecting the discussion here, and feedback from Brian was commented

**Comment by v-karbovnichy** [comment]
Hey team, could you please check this individually or discuss the impact and solution on then next TAG meeting?

**Comment by mitchelsellers** [comment]
@v-karbovnichy We discussed this today during the approvers meeting, we will review this with a proposed recommended path and comment back after our next meeting.

**Comment by berkarslan-xo** [comment]
Any updates on this?

**Comment by v-karbovnichy** [comment]
Hi team, any updates on this?

**Comment by aaiordac** [comment]
Please share updates on this issue
